flink-runtime/src/main/java/org/apache/flink/runtime/state/RegisteredKeyValueStateBackendMetaInfo.java

```diff
@@ -145,13 +146,13 @@ private RegisteredKeyValueStateBackendMetaInfo(
 		return stateSerializerProvider.previousSchemaSerializer();
 	}
 
-	@Nullable
-	public StateSnapshotTransformer<S> getSnapshotTransformer() {
-		return snapshotTransformer;
+	@Nonnull
+	public StateSnapshotTransformFactory<S> getStateSnapshotTransformFactory() {
```

Comment:
Should the factory be called `StateSnapshotTransformerFactory`?

Reply:
True, it can be. This was an attempt to keep it shorter, though not much at the end of the day